src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -544,6 +497,72 @@ void CodeGen::genHWIntrinsic_R_R_RM_I(GenTreeHWIntrinsic* node, instruction ins)
     }
 }
 
+// genHWIntrinsicJumpTableFallback : generate the jump-table fallback for imm-intrinsics
+//                       with non-constant argument
+//
+// Arguments:
+//    nonConstImmReg - the register contains non-constant imm8 argument
+//    baseReg        - a register for the start of the switch table
+//    offsReg        - a register for the offset into the switch table
+//    emitSwCase     - the lambda to geenrate siwtch-case
+//
+// Return Value:
+//    generate the jump-table fallback for imm-intrinsics with non-constant argument.
+// Note:
+//    This function can be used for all imm-intrinsics (whether full-range or not),
+//    The compiler front-end (i.e. importer) is responsible to insert a range-check IR
+//    (GT_HW_INTRINSIC_CHK) for imm8 argument, so this function does not need to do range-check.
+//
+template <typename HWIntrinsicSwitchCaseBody>
+void CodeGen::genHWIntrinsicJumpTableFallback(regNumber                 nonConstImmReg,
+                                              regNumber                 baseReg,
+                                              regNumber                 offsReg,
+                                              HWIntrinsicSwitchCaseBody emitSwCase)
+{
+    assert(nonConstImmReg != REG_NA);
+    emitter* emit = getEmitter();
+
+    const int   maxByte = 256;
```

Comment:
Why is this hardcoded to 256? Shouldn't be 32 for CMPSS instruction? And shifts too have much lower useful maximum. It may be fallback code but IMO it shouldn't be unnecessarily large.

Reply:
Will change.

Reply:
@fiigii, are we differentiating between instructions where an out of range imm
* causes a fault
* is ignored
* is masked

I think we may want to consider how these are handled eventually as causing a fault is the only one we actually need to range check against. If an out of range immediate is ignored, it is simply a nop, and if the upper bits are just masked out, then the input is still valid, even if not "normal" (ex: only bits 0-3 are used and the upper bits are ignored/masked, so imm8=16 and imm8=0 do the same thing).